recipe/build.sh

```diff
@@ -4,34 +4,62 @@ CFLAGS=$(echo "${CFLAGS}" | sed "s/-mtune=[a-zA-Z0-9]*//g")
 # Avoid sorting LDFLAGS
 sed -i.bak 's/LDFLAGS := $(sort $(LDFLAGS))//g' common.mk
 
+
+# Multithreading
+MODEL="pthreads"
+
+
+# Map platform to BLIS target architecture 
+case $target_platform in
+    *-64)
+        arch="x86_64"
```

Comment:
Why are you setting `x86_64` for `osx-64` too?

Reply:
`x86_64` works, is as fast as the `intel64` variant, makes the recipe much cleaner and does not assume anything unnecessary about hardware (who are we to tell people not to use osx on zen? and osx does run on zen). In a nutshell, special-casing is hardly worth it and osx does run on other-than-intel platform so it's nice to have for a tiny cost.

If size is a problem, I have a neat trick to more than halve the size of all those packages: removing the static libraries (or, more accurately, put them in their own package).

Reply:
There's no need to change things in a PR with title "Add aarch64 build support". I would only make this change if there is a request from a user.

Reply:
Besides, non intel64 builds for macos are not supported upstream AFAIK

Reply:
The current script is ugly and unmaintainable. The new one is clean and extensible. It does not break anything therefore it is acceptable refactoring. If things were breaking/regressing, I would have agreed but that not being the case, this is cleaner and my way to add aarch64 support without making things uglier and even less easy to follow.

Reply:
So, your argument to support an unsupported configuration upstream is simply because
```bash
    *-64)
        arch="x86_64"
        ;;
```
is more maintainable than

```bash
    osx-64)
        arch="intel64"
        ;;
    *-64)
        arch="x86_64"
        ;;
```
?

Reply:
No, I made the argument that the current script (with or without the macos special case) is more maintainable and cleaner that the original one. The very fact you can confidently add a few characters *like this* and suppose it's gonna do the right thing (compared to adding a big ugly block of partially copy-pasted code from before) is the best proof I am correct.

> your argument to support an unsupported configuration upstream

Oh hang on. Which upstream are we talking about? 

- macos-64 supports all x86 (there are hackintosh out there proving it's the case).
- apple? apple does not support BLIS in any way.
- BLIS? Blis's "intel64" target was not meant **for** macos. Proof is it supports intel CPUs that were never used by apple in a mac. So the current blis supports configurations that were never released by apple too.

x86_64 is a superset of intel64. Both are as fast on mac. The only downside to providing it is that it'll take a few hundred kB more.

Looking at a situation from another perspective, with "intel64", we know as a fact non-vendor-provided platforms are going to perform badly. With "x86_64", we as a fact vendor-provided macs will preform as well and we would expect the rest to perform as well.

That's the point of distinguishing CPU architecture (actual BLIS instructiont) from OS (how those instructions are presented in a library file).

I also hope you understand that, besides a handful of laptops, Linux has only been running on non-vendor-provided machines for ever. Most hardware today is not even certified to work with Linux, there is no "upstream support". But BLIS only cares about CPUs and how to present the code in a file.

----

So let's swap the argument then: what reason would there be not to include support for AMD CPUs?